crates/ruff_formatter/src/lib.rs

```diff
@@ -112,7 +122,7 @@ impl std::fmt::Display for IndentStyle {
 #[derive(Clone, Copy, Debug, Eq, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 #[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
-pub struct LineWidth(u16);
+pub struct LineWidth(pub u16);
```

Comment:
Thank you. Could you explain why it is necessary to make the fields public? All the fields are intentionally private (here and in `PyFormatOptions`)

Reply:
I hadn't noticed that there was already a formatter for wasm, I looked at the documentation but didn't find it.
My intention was to export it to the dprint plugin.

Reply:
https://github.com/wasm-fmt/ruff_fmt/blob/60669453962a858bbdc3bf015b2f1771603c9104/crates/ruff_fmt_dprint/src/configuration/configuration.rs#L26

https://github.com/wasm-fmt/ruff_fmt/blob/60669453962a858bbdc3bf015b2f1771603c9104/crates/ruff_fmt_dprint/src/configuration/resolve_config.rs#L36-L42

Reply:
> [wasm-fmt/ruff_fmt@6066945/crates/ruff_fmt_dprint/src/configuration/configuration.rs#L26](https://github.com/wasm-fmt/ruff_fmt/blob/60669453962a858bbdc3bf015b2f1771603c9104/crates/ruff_fmt_dprint/src/configuration/configuration.rs#L26)

You can use `LineWidth::try_from(u16)` to get a line width

> [wasm-fmt/ruff_fmt@6066945/crates/ruff_fmt_dprint/src/configuration/resolve_config.rs#L36-L42](https://github.com/wasm-fmt/ruff_fmt/blob/60669453962a858bbdc3bf015b2f1771603c9104/crates/ruff_fmt_dprint/src/configuration/resolve_config.rs#L36-L42)

It may be a bit more cumbersome but you can do

```rust
let resolved_config = PyFormatOptions::default()
	.with_indent_style(indent_style)
	.with_line_with(line_with)
	.with_quote_style(quote_style)
	with_magic_trailing_comma(magic_trailing_comma)
```

You may also want to use `PyFormatOptions::from_extension()` or `::from_source_type`

Reply:
https://github.com/astral-sh/ruff/blob/ac5c8bb3b683fb009b39098c00f9357e9aeb6e38/crates/ruff_python_formatter/src/options.rs#L69-L72

I suppose the signature of `with_quote_style` should be `mut self` and not a reference type. Is that correct?

Reply:
That's correct